### ISSUE TITLE
Queens

### ISSUE THREAD

**Issue by JeffreyStrahm** [siggame/MegaMinerAI-Game-Ideas#35]
I thought about copy and pasting my old suggestion, but it's too different. Let's start from the 
beginning:
Version **2.1**

# **Story**
You are part of a super mutant ant colony with many queens and bizarre ants. It is now just after a harsh winter that you have slept through, and it is time to expand! You are one of the queens that headed out to found a new colony and find the perfect spot. Your joy is short lived, however, as you smell something in the air, your enemies have arrived. Turns out you weren't the only one interested in making a new colony as an enemy mutant ant queen is digging in just a few meters away! It's time to fight for your survival and kill the queen before winter comes again!

# **TLDR**
- Goal: Kill enemy queen. _see **Victory**_
- Turn-based movement. _see **Movement**_
- Map have space around the queen but is filled in with varying dirt and food. _see **Map**_
- Queen generates food. _see **Start & Food**_
- 5 unit types _see **Units**_
   1. **Spawnling** fast, week, naked, free (upgraded into the other unit types)
   2. **Major** slow, strong, armored, expensive (beats Alate)
   3. **Spitter** slow, corrosive ranged, naked, expensive (beats Major )
   4. **Alate** fast, strong, naked, expensive (beats Spitter)
   5. **Queen**, crawl, Nuking, naked, irreplaceable.
- When you generate a unit, place it on one of the eight unoccupied tiles around the queen _see **Units**_
- Upgrading units costs food, the costs increases the more you have. _see **Units & food**_
- Food, dirt, and eggs can be carried by ants  _see **Units**_

# **The Game**
## **Start**
Each Player starts with 5 food, and generates 1 per turn.
The map is procedurally generated, with queens placed on opposite sides.
You won't spawn in the middle

## **Map**
The map will be filled with dirt wall with procedurally generated HP levels (including 0) and food tiles will spawn with just 1 food on it. The queens will get a "round" cavern with the eight spaces around the queen open and any space adjacent to them. These spaces possess no dirt. (Max generated dirt wall HP is 5)
adjacent means to north, south, east, or west of that tile.
The map will be 30 by 30
each tile can possess
 - Dirt wall (can be on the same tile as dirt but not food)
     - Each dirt in a wall raises its HP by one
     - No max HP limit. I would give it 10 shades for texture.
 - dirt (dropped when a wall is dug and can be repacked into a wall.) (No limit to the number on a tile)
 - ant
 - egg
 - food (no limit on a tile, can be on the same tile as ants and eggs)
   - if there is 1 food it will increase by 1 food in 4 turns
   - if there is 2 food it will increase by 1 food in 2 turns
   - if there is 3 food it won't grow.

## **Movement**
There can only one ant per tile. Ants cannot be on tiles that are dirt walls or possess eggs. Even though they can pick up eggs. They can be on the same tile as dirt and food, however. Ants can move as many tiles as their AP(Action Point) attribute dictates per turn. An ant with a 1 can move 1 tile. They don't need to use it all at once, but can only move on their turn.

### **Eggs**
Whenever a unit is being upgraded or healing, an egg is formed. Each ant will take so many turns to "hatch" from the egg. The time it takes for an ant to hatch is dictated by each ant. These eggs can be moved around freely and do not advance when being carried. They advance at the end of every turn. They will also be instantly destroyed if attacked. An ant will be in an egg for 1 turn when healing.

### **Ants**
(AP = action points)
(Stats cannot go below 0, except HP)
If a units HP drops to 0 or below, they die.
The Queen starts with 30 larva
larva replenish at a rate of 1 per 3 turns and do not exceed the population cap.
There is a population cap of 30 (larva count + existing units, not including Queen)

**Spawnling** fast, week, naked, free (swarm)
 - AP: 4
 - Health: 3
 - attack: 2
 - Armor: 0

**Major** slow, strong, armored, expensive (beats Alate)
 - AP: 2
 - Health: 4
 - attack: 4
 - Armor: 1
 - cost: 2
 - Hatch Time: 4

**Spitter** slow, corrosive ranged, naked, expensive (beats Major )
 - AP: 2
 - Health: 3
 - attack: 3
 - range: 3
 - Armor: 0
 - cost: 2
 - Hatch Time: 4
 - Ability - each attack dropps a opponents armor by 1 (affects ajacent cells)

**Alate** fast, strong, naked, expensive (beats Spitter)
 - AP: 4
 - Health: 4
 - attack: 3
 - Armor: 0
 - cost: 2
 - Hatch Time: 4

**Queen**, crawl, Nuking, naked, irreplaceable.
 - AP: 1
 - Health: 200
 - attack: 5
 - range: 5
 - Armor: 0
 - Ability - each attack dropps a opponents armor by 1 (affects ajacent cells)

The cost of all specialized units increases by 1 per specialized unit that exists.

### **Actions**
**Move**
An ant may move to a legal adjacent tile. Costs one AP.

**Attack**
For units with no range, pick an adjacent tile, and attack it. You can only attack once a turn.
For ranged units, they can attack that many "steps" away. Counting range is done the same way as moving. They cannot attack through walls. They can attack empty cells.
Ranged ants can attack walls and lower their HP by 1, dropping a dirt on the ground toward them. An odd range stops ties from occurring.

The damage done is the attack state minus the armor stat. The new total is subtracted from the HP.

**Pickup**
Pick up an item from an adjacent cell or the tile the unit is on. Costs 1 AP.

**drop**
Place an item at the feet of the unit or on an adjacent tile. Costs 1 AP.

**Give**
Place the item you are holding the inventory of an adjacent ant that isn't holding an item. Costs 1 AP.

**Dig**
Lower the HP of an adjacent dirt wall by 1 and add 1 dirt to the tile you stand on. Costs 1 AP.

**Build**
Use the dirt that the ant is holding to increase the HP of a dirt wall on an adjacent tile. If the ant isn't carrying dirt, it can use dirt on the ground. Costs 1 AP.

**Restore**
An ant can revert to its egg form and will hatch in 1 turn, completely restored. Armor included.
Yes, the queen can do this as well.

**Upgrade**
A spawnling at the cost of the food required will turn into an egg and hatch in the dictated number of turns as that unit, fully healed.

**spawn**
The queen deploys one larva as a spawnling in one of the eight tiles around her.

## **Food**
The Queen ant generates 1 food per turn and can store unlimited food. The ants food stores are treated as her carrying the food. She cannot carry dirt. The give action is all that is required for the Queen to dispense food and receive food. The drop command is all that is needed to plant food. The pickup command is all that is needed to harvest food.

## **Victory**
To win you must kill the Queen.
The games time limit is 200 turns when winter arrives and the ants must Hibernate (yes, ant Hibernate)
In that case, the Queen who took the least damage wins
If they are tied, the Queen with the highest total values of units defeated wins (Stolen units count)
If they are still tied, flip a coin.
